src/umpire/strategy/DynamicPoolMap.cpp

```diff
@@ -253,25 +255,26 @@ void DynamicPoolMap::coalesce()
   // size it released
   UMPIRE_REPLAY("\"event\": \"coalesce\", \"payload\": { \"allocator_name\": \"" << getName() << "\" }");
 
-  do_coalesce();
+  do_coalesce(getActualSize());
 }
 
-void DynamicPoolMap::do_coalesce()
+void DynamicPoolMap::do_coalesce(std::size_t suggested_size)
 {
   mergeFreeBlocks();
   // Now all possible the free blocks that could be merged have been
 
   // Only release and create new block if more than one block is present
   if (m_free_map.size() > 1) {
-    const std::size_t released_bytes{releaseFreeBlocks()};
-    // Deallocated and removed released_bytes from m_free_map
+    releaseFreeBlocks(); // Deallocated and removed released_bytes from m_free_map
+    std::size_t size_post{getActualSize()};
 
     // If this removed anything from the map, re-allocate a single large chunk
     // and insert to free map
-    if (released_bytes > 0) {
-      UMPIRE_LOG(Debug, "coalescing " << released_bytes << " bytes.");
-      Pointer ptr{allocateBlock(released_bytes)};
-      insertFree(ptr, released_bytes, true, released_bytes);
+    if (size_post < suggested_size) {
+      std::size_t alloc_size{suggested_size - size_post};
```

Comment:
Do you think we should check if alloc_size is reasonable?

Reply:
Good question. What is reasonable?

Reply:
I think it should be _at least_ the minimum block size (I think that will happen anyway, but good to check)

Reply:
@davidbeckingsale, I am assuming we resolved this in our discussion yesterday.  Please let me know if you are still looking for changes.  Thx

Reply:
I think the decision was made. I don't think we need to enforce anything, but can you please add a log message here that prints the alloc_size to Debug?

Reply:
Do you mean like on line 275?